### PR TITLE
Allow axes to be constructed by namedtuple syntax in Map

### PR DESCRIPTION
**Pull request by asinghvi17**
Allows 

```julia
b = Rect2f(-20.0, -20.0, 40.0, 40.0)
m = Tyler.Map(b, provider=OpenStreetMap(), axis = (; type = GeoAxis))

f = Figure()
m = Tyler.Map(b, provider = OpenStreetMap(), figure = f[1, 2])
```

### CODE REVIEW

File: src/map.jl [hunk 1]
```diff
@@ -58,6 +58,56 @@ struct Map{Ax<:Makie.AbstractAxis} <: AbstractMap
...
+    AxisType = pop!(axis_kws, :type, Axis)
+
+    axis = AxisType(figure[1, 1]; axis_kws...)
+
+    setup_axis!(axis, ext_target, crs)
+
+    return figure, axis
+end
+
+_get_parent_layout(gp::Makie.GridPosition) = _get_parent_layout(gp.layout)
```
Comment (rafaqz) [inline src/map.jl:77]: Feel like these helpers could go in Makie.jl eventually?

**Review by rafaqz — APPROVED** [review]
LGTM, I wanted this the other day instead of setting up a `GeoAxis` manually

**Comment by lazarusA** [comment]
One test is failing 😔

File: src/map.jl (hunk 1 above)
Comment (asinghvi17) [reply src/map.jl:77]: Yeah they should eventually...for now it's mergeable though

File: test/runtests.jl [hunk 2]
```diff
@@ -38,6 +38,19 @@ display(m)
     @test GeoInterface.crs(m) == Tyler.MapTiles.WebMercator()
 end
 
+@testset "NamedTuple axis syntax" begin
+    b = Rect2f(-20.0, -20.0, 40.0, 40.0)
+    m = @test_nowarn Tyler.Map(b, provider=Tyler.TileProviders.OpenStreetMap(), axis = (; type = Axis, aspect = AxisAspect(1)))
+    @test only(contents(m.figure.layout[1, 1])) isa Axis
+    @test only(contents(m.figure.layout[1, 1])).aspect == AxisAspect(1)
```
Comment (lazarusA) [inline test/runtests.jl:45]: almost, I think here, we need to call-in the value from the `Observable` coming out of `only`. And maybe similarly for the next test as well.

Comment (asinghvi17) [reply test/runtests.jl:45]: True... not sure how i missed that!

**Comment by asinghvi17** [comment]
This needs a rebase, let's not merge yet

**Comment by asinghvi17** [comment]
Ok...I have no clue what's going on with this segfault :(

**Comment by asinghvi17** [comment]
We could also add an LMap, as a way to switch away from requiring control over the figure...

**Comment by asinghvi17** [comment]
It looks like the segfault is a known error in the interaction between julia and curl: 

https://github.com/curl/curl/issues/14860

so I think we can merge as is.  Doc build failed for external reasons (I think)...

**Comment by lazarusA** [comment]
mmm... but is only happening in this PR. And now in master is also failing.

**Comment by giordano** [comment]
> It looks like the segfault is a known error in the interaction between julia and curl:

That doesn't make sense, we don't use curl 8.10+ yet.

**Comment by asinghvi17** [comment]
You're right, it was another reason (non-initialized field of a mutable struct).

**Pull request merged**